### ISSUE TITLE
Rule Change: make `no-irregular-whitespace` catch `\u3164` as well

### ISSUE THREAD

**Issue by koskokos2** [eslint/eslint#15281]
### What rule do you want to change?

no-irregular-whitespace

### What change to do you want to make?

Generate more warnings

### How do you think the change should be implemented?

A new default behavior

### Example code

```js
// https://certitude.consulting/blog/wp-content/uploads/2021/11/hangul_filler_backdoor.png
const checkCommands = [
  'ping -c 1 google.com',
  'curl -s http://example.com/',\u3164
];
```

### What does the rule currently do for this code?

Not taking `u3164` into account.

### What will the rule do after it's changed?

Warn/error about the `u3164`.

### Participation

- [X] I am willing to submit a pull request to implement this change.

### Additional comments

Based on this article https://certitude.consulting/blog/en/invisible-backdoor/

**Comment by nzakas** [comment]
I think we are covering this topic in this discussion: https://github.com/eslint/eslint/issues/15240

**Comment by koskokos2** [comment]
Yes, it is quite similar. Although I cannot find this specific Unicode char \u3164 in there. You might be right to merge this with the issue you mentioned. But in the same time I feel it fits quite well into existing rule I proposed to change.

**Comment by koskokos2** [comment]
Based on this comment https://news.ycombinator.com/item?id=29172909 there is also `U+FFA0` char.

**Comment by nzakas** [comment]
@eslint/eslint-tsc thoughts?

**Comment by koskokos2** [comment]
That link results in 404, but I think it meant to be https://github.com/unlight/eslint-plugin-tsc. But not sure I understand how it is related to this issue.

**Comment by snitin315** [comment]
> That link results in 404, but I think it meant to be https://github.com/unlight/eslint-plugin-tsc. But not sure I understand how it is related to this issue.

No, it wasn't meant to be https://github.com/unlight/eslint-plugin-tsc

It is a private link that is accessible to the ESLint team only. Here, we are just sending a ping to the [`Technical Steering Committee`](https://github.com/eslint/eslint#technical-steering-committee-tsc) members for their input.

**Comment by btmills** [comment]
#15240 proposes catching bidi characters and homoglyphs. It sounds like we should also catch a third category of invisible characters like `\u3164` and `\uFFA0`. `no-irregular-whitespace` already catches a bunch of invisibles. However, it looks like the Hangul Filler is considered a _letter_. Would we still want to cover it in `no-irregular-whitespace` under the philosophy that invisible characters are a lot like whitespace even if this one is technically a letter?

**Comment by nzakas** [comment]
Good point. We should probably not include letters, even if they are invisible, in a white space-focused rule. 

Given that this is presented as a security issue, would such a rule be better for a security plugin?

**Comment by nzakas** [comment]
Keeping open for transitioning to a security plugin.

**Comment by nzakas** [comment]
Ported to eslint-plugin-security here:
https://github.com/eslint-community/eslint-plugin-security/issues/116
